src/adapters/carbon-defi/products/strategies/carbonDeFiStrategiesAdapter.ts

```diff
@@ -0,0 +1,499 @@
+import { BigNumber } from 'bignumber.js'
+import { BigNumberish } from 'ethers'
+import { SimplePoolAdapter } from '../../../../core/adapters/SimplePoolAdapter'
+import { AdaptersController } from '../../../../core/adaptersController'
+import { Chain } from '../../../../core/constants/chains'
+import { NotImplementedError } from '../../../../core/errors/errors'
+import { CustomJsonRpcProvider } from '../../../../core/utils/customJsonRpcProvider'
+import { getTokenMetadata } from '../../../../core/utils/getTokenMetadata'
+import {
+  ProtocolAdapterParams,
+  ProtocolDetails,
+  PositionType,
+  GetPositionsInput,
+  GetEventsInput,
+  MovementsByBlock,
+  GetTotalValueLockedInput,
+  GetApyInput,
+  GetAprInput,
+  GetConversionRateInput,
+  ProtocolTokenApr,
+  ProtocolTokenApy,
+  ProtocolTokenUnderlyingRate,
+  ProtocolTokenTvl,
+  ProtocolPosition,
+  TokenType,
+  TokenBalance,
+  Underlying,
+  UnderlyingTokenRate,
+} from '../../../../types/adapter'
+import { Erc20Metadata } from '../../../../types/erc20Metadata'
+import { Protocol } from '../../../protocols'
+import { CarbonController__factory, Voucher__factory } from '../../contracts'
+import {
+  StrategyCreatedEvent,
+  StrategyDeletedEvent,
+  StrategyUpdatedEvent,
+} from '../../contracts/CarbonController'
+
+const contractAddresses: Partial<
+  Record<
+    Chain,
+    { carbonControllerAddress: string; voucherContractAddress: string }
+  >
+> = {
+  [Chain.Ethereum]: {
+    carbonControllerAddress: '0xC537e898CD774e2dCBa3B14Ea6f34C93d5eA45e1',
+    voucherContractAddress: '0x3660F04B79751e31128f6378eAC70807e38f554E',
+  },
+}
+
+enum StrategyUpdateReason {
+  Edit = '0',
+  Trade = '1',
+}
+
+export class CarbonDeFiStrategiesAdapter extends SimplePoolAdapter {
```

Comment:
Since the methods of SimplePoolAdapter are not being used, it'd be better to `implements IProtocolAdapter` instead of `extends SimplePoolAdapter`.

That way you should also be able to remove the protected abstract methods from `SimplePoolAdapter`, as they are not being used.